docs/extensions/cross-chain-data/example_pt_1.mdx

```diff
@@ -0,0 +1,273 @@
+- [Running a Kwil node locally](/docs/node/quickstart#single-node)
+
+## Step 1. Credit Struct
+
+We will first define a struct to hold the information about our credit. This struct will be deterministically serializable, allowing it to be
+passed to stored in Kwil's resolution store. We will use Kwil's `serialize` package, which implements a deterministic and compact serialization
+format based on Ethereum's RLP encoding.
+
+Create a new directory in the `kwil-db/extensions` directory called `resolution_credit`, and create a new file called `credit.go`.
+
+```go title="credit_resolution.go"
+package main
+
+import (
+	"math/big"
+)
+
+// AccountCreditResolution is a resolution that allows accounts to be credited with a given amount.
+// It is used by both the credit_account resolution and the eth_deposit_oracle.
+// It can be serialized and deserialized to be passed around the network.
+// The amount cannot be negative, as this will fail RLP encoding.
+type AccountCreditResolution struct {
+	// Account is the account to be credited.
+	// This can be an Ethereum address (decoded from hex), a validator ed25519 public key,
+	// or any other custom account identifier implemented in an auth extension.
+	Account []byte
+	// Amount is the amount to be credited to the account.
+	// It uses a big.Int to allow for arbitrary precision, and to allow for uint256 values,
+	// which are commonly used in token contracts on Ethereum.
+	Amount *big.Int
+	// TxHash is the hash of the Ethereum transaction that emitted the EVM event to credit the account.
+	// This ensures that, even if the same account is credited the same amount multiple times,
+	// that each credit resolution is unique. It is critical that all resolutions in Kwil are
+	// unique, as they are idempotent for the lifetime of the entire network.
+	TxHash []byte
+}
+
+// MarshalBinary marshals the AccountCreditResolution to binary.
+// We do not use the popular json.Marshal library because we need this serialization
+// to be deterministic. Kwil contains a serialization library that uses Ethereum's
+// RLP encoding, which is deterministic and used for all serialization in Kwil.
+func (a *AccountCreditResolution) MarshalBinary() ([]byte, error) {
+	return serialize.Encode(a)
+}
+
+// UnmarshalBinary unmarshals the AccountCreditResolution from binary.
+// It is the inverse of MarshalBinary, and uses the same serialization library.
+func (a *AccountCreditResolution) UnmarshalBinary(data []byte) error {
+	return serialize.DecodeInto(data, a)
+}
+```
+
+## Step 2. Resolution Config
+
+We will now define the configuration for our resolution. This informs the Kwil network about the rules for our resolution, and
+how it should be processed. We will register this resolution using Golang's `init` function. This is a special function called by
+Golang when the package is imported, and will be run before any other code. This allows our `kwild` node to access the resolution
```

Comment:
Should the purpose of init be explained here? You don't explain it in any other sections.

Reply:
```suggestion
how it should be processed. We will register this resolution using Golang's `init` function. This is a special function called by
Golang when the package is imported, and will run before any other code. This allows our `kwild` node to access the resolution
```